From fa61c3dfb2c86e91abcc35464bff13a3678f1e03 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Mon, 30 Nov 2020 20:07:33 +0200
Subject: [PATCH] [SW-20154] Makefile: expand include path for version info

VERSION_STR is extracted from preboot.h file, using a
simple sed script. If the preboot.h file has any other
header inclusion, it will result in an error of "file
not found" since the include path is just limited
to current working directory. To prevent this, expand
the include path to include other folders too.

Change-Id: I6321e02b3f9650efb0c56b0dbca8a0841667e8f8
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/Makefile b/Makefile
index dadb547..92919e3 100644
--- a/Makefile
+++ b/Makefile
@@ -76,7 +76,7 @@
 	@date +'#define PREBOOT_TIME "%T"' >> $@
 
 buildversion-habanalabs: include/preboot.h $(AUTO_GENERATED_FILE)
-	echo "VERSION_STR" | gcc -include $< -I./ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
+	echo "VERSION_STR" | gcc -include $< -I./ -I ./common/ -I ./$(HL_DEV)/ -I$(SPECS_DIR)/$(SPECS_DEV)/asic_reg -I$(SPECS_DIR)/ -I ./include/ -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
 
 $(PCI_PHY_FW_TARGET): $(HL_DEV)/pcie_phy_fw.i
 	@cut -d' ' -f2- $< | xxd -r -p | dd status=none bs=16 count=$$(($(FW_SIZE) / 16)) conv=sync,swab of=$@
